### DISCUSSION TITLE
TestWatcher vs TestExecutionListener

### DISCUSSION THREAD

**Discussion by ghost** [junit-team/junit-framework#3994] [Q&A] [answered]
Please give an explanation when it is better to use one or the other option.

At first glance it looks like listener replaces watcher completely and also has additional features.

**Comment by marcphilipp** [comment, marked as answer]
`TestWatcher` is Jupiter-specific and should be used whenever you want to react based on the outcome of a test (e.g. in combination with another extension point) or change its behavior. `TestExecutionListener` is a Platform concept and cannot influence the outcome of a test or interact with Jupiter extensions.
